iocore/net/BIO_fastopen.cc

```diff
@@ -27,13 +27,26 @@
 
 #include "BIO_fastopen.h"
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#define BIO_set_data(a, ptr) ((a)->ptr = (ptr))
+#define BIO_get_data(a) ((a)->ptr)
+#define BIO_set_init(a, init) ((a)->init = (init))
+#define BIO_set_shutdown(a, shut) ((a)->shutdown = (shut))
+#define BIO_get_shutdown(a) ((a)->shutdown)
+#endif
+
+struct Data {
+  Data() : fd(NO_FD), dst(nullptr) {}
+  int fd;
+  const sockaddr *dst;
+};
+
 static int
 fastopen_create(BIO *bio)
 {
-  bio->init  = 0;
-  bio->num   = NO_FD;
-  bio->flags = 0;
-  bio->ptr   = nullptr;
+  Data *data = new Data;
```

Comment:
Is this correct? Should I "delete data" in fastopen_destroy()?
Sorry, I'm sketchy on Traffic Server and C++ memory management patterns ...